edx_sga/static/js/src/edx_sga.js

```diff
@@ -292,6 +303,37 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $(self).addClass("disabled");
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(element).find('.task-message').hide();
+                      } else {
+                        $(element).find('.task-message').show();
+                      }
+                      $(self).removeClass("disabled");
+                    }
+                  );
+                });
+
+                setTimeout(function () {
+                  $.get(
+                    downloadSubmissionsStatusUrl,
+                    function(data) {
+                      if (data["zip_available"]) {
+                        $(element).find('.task-message').hide();
+                      } else {
+                        $(element).find('.task-message').show();
+                      }
+                    }
+                  );
+                }, 10000);
```

Comment:
This UI does not make sense. You moved the message to the HTML template, so now every time I bring up the grading modal, I see "Started preparing student submissions zip file...". This is shown even if I haven't clicked the button/link to prepare the submissions zip file. In other words, I haven't asked for a submissions zip file to be prepared at all, and it's not even true that a zip file is being prepared on the back end in this case. 

We should only show that message if a user has clicked the "Download" button link and the zip file is not yet ready. It's best to get this PR merged soon, so in terms of checking whether the file is ready yet, I would be OK with something simple: make requests to `downloadSubmissionsStatusUrl` in a `for` or `while` loop every 3 seconds or so up to some limit (5-10 tries). If the request indicates that the file is ready, we can exit the loop, re-enable the download button, and remove the "Started preparing" message or change it to something like "Student submission file ready for download". We should also change the "Started preparing" message to remove the text saying "Come back later".

Reply:
@gsidebo max file size can be 10 mbs, assignments record can be equal to 3gb